ledger/db/src/lib.rs

```diff
@@ -676,14 +670,17 @@ impl LedgerDB {
 
         // Check that none of the key images were previously spent.
         for key_image in &block_contents.key_images {
-            if self.contains_key_image(key_image)? {
+            if self
```

Comment:
Previously, this function used two database transactions to perform the key images and output uniqueness checks. That was incorrect - all reads should happen inside a single transaction to ensure we are looking at a consistent snapshot of the ledger.